### ISSUE TITLE
Error connecting to db2 when Security=SSL

### ISSUE THREAD

**Issue by dlema65** [ibmdb/python-ibmdbsa#114]
Good day:

I am not sure if the problem is with this package or with ibm-db, but when trying to connect to db2 using
```
from sqlalchemy import create_engine

create_engine('db2+ibm_db://****:****@my_db2_instance.cloud.ibm.com:50001/mydb?Security=SSL')
```
it fails in .../ibm_db_sa/ibm_db.py line 200 with error TypeError: 'sqlalchemy.cimmutabledict.immutabledict' object does not support item deletion.

When connecting with

`create_engine('db2+ibm_db://****:****@my_db2_instance.cloud.ibm.com:50001/mydb;Security=SSL;)`

it works, but that ";Security=SSL;" is not valid in URL and the problem is that 
```
sqlalchemy.engine.URL.create(
            driver,
            username=uid,
            password=pwd,
            host=server,
            port=port,
            database=database,
            query={'Security': 'SSL'}
)
```
will create a standard URL, as in the first sample instead of the second one.

Thank you so much.

**Comment by bchoudhary6415** [comment]
Hello @dlema65 
can you please refer below connection string
{
  "hostname": "db2w-host.us-south.db2w.cloud.ibm.com",
  "password": "passwd",
  "https_url": "https://db2w-host.us-south.db2w.cloud.ibm.com",
  "port": 50001,
  "ssldsn": "DATABASE=BLUDB;HOSTNAME=db2w-host.us-south.db2w.cloud.ibm.com;PORT=50001;PROTOCOL=TCPIP;UID=bluadmin;PWD=passwd;Security=SSL;",
  "host": "db2w-host.us-south.db2w.cloud.ibm.com",
  "uri": "db2://bluadmin:passwd@db2w-host.us-south.db2w.cloud.ibm.com:50001/BLUDB?ssl=true;",
  "db": "BLUDB",
  "username": "bluadmin",
  "ssljdbcurl": "jdbc:db2://db2w-host.us-south.db2w.cloud.ibm.com:50001/BLUDB:sslConnection=true;"
}


For Sql Alchemy can you need to use the URI syntax for connection.
Also while providing the URI
"uri": "db2://bluadmin:[passwd@db2w-host.us-south.db2w.cloud.ibm.com](mailto:passwd@db2w-host.us-south.db2w.cloud.ibm.com):50001/BLUDB?ssl=true;",

you can please check after changing the dialect from db2 to ibm_db_sa
ibm_db_sa://bluadmin:[passwd@db2w-host.us-south.db2w.cloud.ibm.com](mailto:passwd@db2w-host.us-south.db2w.cloud.ibm.com):50001/BLUDB?ssl=true

can you try once with the above mentioned URI?

Please also give a try by changing the ssl=true to SECURITY=SSL
ibm_db_sa://bluadmin:[passwd@db2w-host.us-south.db2w.cloud.ibm.com](mailto:passwd@db2w-host.us-south.db2w.cloud.ibm.com):50001/BLUDB?SECURITY=SSL

**Comment by bchoudhary6415** [comment]
Hello @dlema65 

This is the known issue and already fixed and fixed commit is [22226c68581f4489d25a872e2ca990fd1710dae7](https://github.com/ibmdb/python-ibmdbsa/commit/22226c68581f4489d25a872e2ca990fd1710dae7)

The solution will be delivered by next release 0.3.9 

Thank you

**Comment by bchoudhary6415** [comment]
Hello @dlema65 
ibm_db_sa v0.3.9 is released now.
Can you please check for error now?
if it get resolved, we can close the issue.
